website/docs/plugin/framework/deprecations.mdx

```diff
@@ -0,0 +1,1282 @@
+	resp.Schema = schema.Schema{
+		// ... other configuration ...
+
+		Attributes: map[string]schema.Attribute{
+			// ... other attributes ...
+
+			"existing_attribute": schema.StringAttribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+type exampleWidgetResourceData struct {
+	// ... other attributes ...
+
+	ExistingAttribute types.String `tfsdk:"existing_attribute"`
+}
+
+func (e *exampleWidgetResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data exampleWidgetResourceData
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// ... other logic ...
+	data.ExistingAttribute = // set to computed value
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (e *exampleWidgetResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data exampleWidgetResourceData
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// ... other logic ...
+	data.ExistingAttribute = // set to computed value
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (e *exampleWidgetResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data exampleWidgetResourceData
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// ... other logic ...
+	data.ExistingAttribute = // set to computed value
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (e *exampleWidgetResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// ... other logic ...
+}
+```
+
+In order to support renaming `existing_attribute` to `new_attribute`, this sample can be written as the following to support both attributes simultaneously until the `existing_attribute` is removed:
+
+```go
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ resource.Resource = (*exampleWidgetResource)(nil)
+
+type exampleWidgetResource struct{}
+
+func NewWidgetResource() resource.Resource {
+	return &exampleWidgetResource{}
+}
+
+func (e *exampleWidgetResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_widget"
+}
+
+func (e *exampleWidgetResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// ... other configuration ...
+
+		Attributes: map[string]schema.Attribute{
+			// ... other attributes ...
+
+			"existing_attribute": schema.StringAttribute{
+				Computed:           true,
+				DeprecationMessage: "use new_attribute instead",
```

Comment:
Is it worth pointing out that `DeprecationMessage` on computed-only attributes are not surfaced through the CLI?

Reply:
I can add some explanation and maybe a link to the schema/computed docs: https://developer.hashicorp.com/terraform/plugin/framework/handling-data/schemas#deprecationmessage-1

Are `DeprecationMessage`s for `Computed` fields **never** surfaced to the CLI as a warning? This doc above combines the messaging with a note about earlier terraform versions and `Required`/`Optional`, but they seem to be separate notes.

Reply:
> Are DeprecationMessages for Computed fields never surfaced to the CLI as a warning?

I believe that's correct but perhaps @bflad would like to comment?

Reply:
The [`DeprecationMessage` field Go documentation](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.1.1/resource/schema#BoolAttribute.DeprecationMessage) tries to go into more detail here:

```
	// In Terraform 1.2.7 and later, this warning diagnostic is displayed any
	// time a practitioner attempts to configure a value for this attribute and
	// certain scenarios where this attribute is referenced.
	//
	// In Terraform 1.2.6 and earlier, this warning diagnostic is only
	// displayed when the Attribute is Required or Optional, and if the
	// practitioner configuration sets the value to a known or unknown value
	// (which may eventually be null). It has no effect when the Attribute is
	// Computed-only (read-only; not Required or Optional).
	//
	// Across any Terraform version, there are no warnings raised for
	// practitioner configuration values set directly to null, as there is no
	// way for the framework to differentiate between an unset and null
	// configuration due to how Terraform sends configuration information
	// across the protocol.
	//
	// Additional information about deprecation enhancements for read-only
	// attributes can be found in:
	//
	//  - https://github.com/hashicorp/terraform/issues/7569
	//
```

Reply:
Thanks for the context! I added a blurb in the beginning of the documentation noting to reference the schema docs for more info on when warnings will be raised: [16f88fe](https://github.com/hashicorp/terraform-plugin-framework/pull/684/commits/16f88fe2b16e54aab3d154397b5bf3c2d464debe)